python/grass/pygrass/utils.py

```diff
@@ -436,6 +436,432 @@ def table_exist(cursor, table_name):
     return bool(one and one[0])
 
 
+def txt2numpy(
+    tablestring,
+    sep=",",
+    names=None,
+    null_value=None,
+    fill_value=None,
+    comments="#",
+    usecols=None,
+    encoding=None,
+    structured=True,
+):
+    """Read table-like output from grass modules as Numpy array;
+    format instructions are handed down to Numpys genfromtxt function
+
+    :param stdout: tabular stdout from GRASS GIS module call
+    :type stdout: str|byte
+
+    :param sep: Separator delimiting columns
+    :type sep: str
+
+    :param names: List of strings with names for columns
+    :type names: list
+
+    :param null_value: Characters representing the no-data value
+    :type null_value: str
+
+    :param fill_value: Value to fill no-data with
+    :type fill_value: str
+
+    :param comments: Character that identifies comments in the input string
+    :type comments: str
+
+    :param usecols: List of columns to import
+    :type usecols: list
+
+    :param structured: return structured array if True, un-structured otherwise
+    :type structured: bool
+
+    :return: numpy.ndarray
+
+        >>> import grass.script.core as grasscore
+        >>> import numpy as np
+        >>> txt = grasscore.read_command(
+        ...     "r.stats", flags="cn", input="basin_50K,geology_30m", separator="|"
+        ... )
+        >>> np_array = txt2numpy(txt, sep="|", names=None)
+        >>> print(np_array)
+
+    """
+
+    from io import BytesIO
+    import numpy as np
+
+    if not encoding:
+        encoding = grassutils._get_encoding()
+
+    if type(tablestring).__name__ == "str":
+        tablestring = grasscore.encode(tablestring, encoding=encoding)
+    elif type(tablestring).__name__ != "bytes":
```

Comment:
Why not testing the types or instances (isinstance) instead of names as strings?